react/components/Account/components/BaseLoading.js

```diff
@@ -0,0 +1,63 @@
+import React, { Component, Fragment } from 'react'
+import PropTypes from 'prop-types'
+
+import ReloadableError from './ReloadableError'
+import ContentWrapper from './ContentWrapper'
+
+class BaseLoading extends Component {
+  state = {
+    isLoading: true,
+  }
+
+  componentDidUpdate = (prevProps) => {
+    if (prevProps.queryData.loading !== this.props.queryData.loading) {
+      this.setState({ isLoading: false })
+    }
+  }
+
+  handleReload = () => {
+    this.setState({ isLoading: true })
+
+    this.props.queryData.refetch().catch(() => {
+      this.setState({ isLoading: false })
+    })
+  }
+
+  render() {
+    const { isLoading } = this.state
+    const { headerConfig, children, queryData } = this.props
+
+    const hasAuthenticationError =
+      queryData.error &&
+      queryData.error.toString().indexOf('not authenticated') > -1
+
+    return (
+      <ContentWrapper {...headerConfig}>
+        {() => (
+          <Fragment>
```

Comment:
Why do you need this?

Reply:
https://github.com/vtex-apps/my-account/blob/085da0677e9bec38ce12b5127fda5823be57850d/react/components/loaders/AddressCreateLoading.js#L9